e2e/lib/k8up.bash

```diff
@@ -19,13 +30,56 @@ kustomize() {
 	go run sigs.k8s.io/kustomize/kustomize/v3 "${@}"
 }
 
+restic() {
+	kubectl run wrestic \
+		--rm \
+		--attach \
+		--restart Never \
+		--namespace "${DETIK_CLIENT_NAMESPACE-"k8up-system"}" \
+		--image "${WRESTIC_IMAGE-quay.io/vshn/wrestic}" \
+		--env "AWS_ACCESS_KEY_ID=myaccesskey" \
+		--env "AWS_SECRET_KEY=mysecretkey" \
+		--env "RESTIC_PASSWORD=myreposecret" \
+		--pod-running-timeout 10s \
+		--quiet=true \
+		--command -- \
+			restic \
+			--no-cache \
+			-r "s3:http://minio.minio.svc.cluster.local:9000/backup" \
+			"${@}" \
+			--json
+}
+
 prepare() {
 	mkdir -p "debug/${1}"
 	kustomize build "${1}" -o "debug/${1}/main.yml"
-	sed -i -e "s|\$E2E_IMAGE|'${E2E_IMAGE}'|" "debug/${1}/main.yml"
+	sed -i \
+		-e "s|\$E2E_IMAGE|'${E2E_IMAGE}'|" \
+		"debug/${1}/main.yml"
+	sed -i \
+		-e "s|\$WRESTIC_IMAGE|'${WRESTIC_IMAGE}'|" \
+		"debug/${1}/main.yml"
 }
 
 apply() {
 	prepare "${@}"
 	kubectl apply -f "debug/${1}/main.yml"
 }
+
+given_a_subject() {
+	kubectl create namespace k8up-e2e-subject || true
+	apply definitions/subject
+}
+
+given_s3_storage() {
+	# TODO HELM REPO
+	helm install minio \
```

Comment:
Don't forget to also install this in the e2e GH workflow, e.g. with https://github.com/Azure/setup-helm

Reply:
actually, we might not need to do this.
The GH machine might already have this: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-README.md#package-management. any helm v3 should be fine for us.